apis/dns/v1alpha1/dns_types.go

```diff
@@ -0,0 +1,397 @@
+	ARecords []ARecord `json:"aRecords,omitempty"`
+
+	// AaaaRecords - The list of AAAA records in the record set.
+	// +optional
+	AaaaRecords []AaaaRecord `json:"aaaaRecords,omitempty"`
+
+	// MxRecords - The list of MX records in the record set.
+	// +optional
+	MxRecords []MxRecord `json:"mxRecords,omitempty"`
+
+	// NsRecords - The list of NS records in the record set.
+	// +optional
+	NsRecords []NsRecord `json:"nsRecords,omitempty"`
+
+	// PtrRecords - The list of PTR records in the record set.
+	// +optional
+	PtrRecords []PtrRecord `json:"ptrRecords,omitempty"`
+
+	// SrvRecords - The list of SRV records in the record set.
+	// +optional
+	SrvRecords []SrvRecord `json:"srvRecords,omitempty"`
+
+	// TxtRecords - The list of TXT records in the record set.
+	// +optional
+	TxtRecords []TxtRecord `json:"txtRecords,omitempty"`
+
+	// CnameRecord - The CNAME record in the  record set.
+	// +optional
+	CnameRecord CnameRecord `json:"cnameRecord,omitempty"`
+
+	// SoaRecord - The SOA record in the record set.
+	// +optional
+	SoaRecord SoaRecord `json:"soaRecord,omitempty"`
+
+	// CaaRecords - The list of CAA records in the record set.
+	// +optional
+	CaaRecords []CaaRecord `json:"caaRecords,omitempty"`
+}
+
+// RecordSetObservation define the actual state of an Azure DNS RecordSet.
+type RecordSetObservation struct {
+	// ID - Resource ID
+	ID string `json:"id,omitempty"`
+
+	// Etag - The etag of the zone.
+	Etag string `json:"etag,omitempty"`
+
+	// Name - The name of the zone.
+	Name string `json:"name,omitempty"`
+
+	// Type - Resource type.
+	Type string `json:"type,omitempty"`
+
+	// Fqdn - Fully qualified domain name of the record set.
+	Fqdn string `json:"fqdn,omitempty"`
+	// ProvisioningState -provisioning State of the record set.
+	ProvisioningState string `json:"provisioningState,omitempty"`
+}
+
+// ARecord an A record.
+type ARecord struct {
+	// Ipv4Address - The IPv4 address of this A record.
+	Ipv4Address string `json:"ipv4Address,omitempty"`
+}
+
+// AaaaRecord an AAAA record.
+type AaaaRecord struct {
+	// Ipv6Address - The IPv6 address of this AAAA record.
+	Ipv6Address string `json:"ipv6Address,omitempty"`
+}
+
+// CaaRecord a CAA record.
+type CaaRecord struct {
+	// Flags - The flags for this CAA record as an integer between 0 and 255.
+	Flags int `json:"flags,omitempty"`
+	// Tag - The tag for this CAA record.
+	Tag string `json:"tag,omitempty"`
+	// Value - The value for this CAA record.
+	Value string `json:"value,omitempty"`
+}
+
+// CnameRecord a CNAME record.
+type CnameRecord struct {
+	// Cname - The canonical name for this CNAME record.
+	Cname string `json:"cname,omitempty"`
+}
+
+// MxRecord an MX record.
+type MxRecord struct {
+	// Preference - The preference value for this MX record.
+	Preference int `json:"preference,omitempty"`
+	// Exchange - The domain name of the mail host for this MX record.
+	Exchange string `json:"exchange,omitempty"`
+}
+
+// NsRecord an NS record.
+type NsRecord struct {
```

Comment:
```suggestion
type NSRecord struct {
```
And also similar capitalization for other type names below.